test/utils/test_utils.py

```diff
@@ -83,35 +83,16 @@ def test_get_regularization(regularization, lam, expected):
     assert result == expected
 
 
-@pytest.mark.parametrize("regularization", ["l0", "l1", "l2"])
-@pytest.mark.parametrize("lam", [1, np.array([1]), np.array([[1]])])
-def test_get_prox_and_regularization_shape(regularization, lam):
-    data = np.array([[-2, 5]]).T
-    reg = get_regularization(regularization)
-    reg_result = reg(data, lam)
-    prox = get_prox(regularization)
-    prox_result = prox(data, lam)
-    assert reg_result is not None
-    assert prox_result is not None
-
-
-@pytest.mark.parametrize(
-    "regularization", ["weighted_l0", "weighted_l1", "weighted_l2"]
-)
-@pytest.mark.parametrize("lam", [np.array([[1, 2]]).T])
-def test_get_weighted_prox_and_regularization_shape(regularization, lam):
-    data = np.array([[-2, 5]]).T
-    reg = get_regularization(regularization)
-    reg_result = reg(data, lam)
-    prox = get_prox(regularization)
-    prox_result = prox(data, lam)
-    assert reg_result is not None
-    assert prox_result is not None
-
-
 @pytest.mark.parametrize("regularization", ["l0", "l1", "l2"])
 @pytest.mark.parametrize(
-    "lam", [np.array([[1, 2]]), np.array([1, 2]), np.array([[1, 2]]).T]
+    "lam",
+    [
+        np.array([[1, 2]]),
+        np.array([1, 2]),
+        np.array([[1, 2]]).T,
+        np.array([1]),
+        np.array([[1]]),
+    ],
```

Comment:
I don't think we need so many test cases.  Try just (a) a size-1 array and (b) an array the same shape as input